src/acpi.rs

```diff
@@ -419,3 +424,357 @@ pub fn process_msct() -> (
+            0,
+            "Not multiple of page-size."
+        );
+        mem_desc.ty = MemoryType::PERSISTENT_MEMORY;
+        mem_desc.phys_start = (*entry).Address;
+        mem_desc.page_count = (*entry).Length / BASE_PAGE_SIZE as u64;
+        mem_desc.att = MemoryAttribute::from_bits_truncate((*entry).MemoryMapping);
+
+        mem_desc
+    }
+}
+
+// NFIT subtable type = 0x1
+fn parse_nfit_region_mapping_structure(entry: *const ACPI_NFIT_MEMORY_MAP) {
+    // To parse Flags
+    const ACPI_NFIT_MEM_SAVE_FAILED: u32 = 0x1; /* 00: Last SAVE to Memory Device failed */
+    const ACPI_NFIT_MEM_RESTORE_FAILED: u32 = 0x1 << 1; /* 01: Last RESTORE from Memory Device failed */
+    const ACPI_NFIT_MEM_FLUSH_FAILED: u32 = 0x1 << 2; /* 02: Platform flush failed */
+    const ACPI_NFIT_MEM_NOT_ARMED: u32 = 0x1 << 3; /* 03: Memory Device is not armed */
+    const ACPI_NFIT_MEM_HEALTH_OBSERVED: u32 = 0x1 << 4; /* 04: Memory Device observed SMART/health events */
+    const ACPI_NFIT_MEM_HEALTH_ENABLED: u32 = 0x1 << 5; /* 05: SMART/health events enabled */
+    const ACPI_NFIT_MEM_MAP_FAILED: u32 = 0x1 << 6; /* 06: Mapping to SPA failed */
+
+    unsafe {
+        assert_eq!((*entry).Header.Type, 1);
+        debug!(
+            "ACPI_NFIT_TYPE_MEMORY_MAP - Type {}, Length {}
+                NfitDeviceHandle: 0x{:x},
+                NfitDeviceHandle.DimmNumber: 0x{:x},
+                NfitDeviceHandle.MemChannel: 0x{:x},
+                NfitDeviceHandle.MemControllerId: 0x{:x},
+                NfitDeviceHandle.SocketId: 0x{:x},
+                NfitDeviceHandle.NodeControllerId: 0x{:x},
+                PhysicalId: {},
+                RegionId: {},
+                RangeIndex: {},
+                RegionIndex: {},
+                RegionSize: {},
+                RegionOffset: {},
+                Address: {},
+                InterleaveIndex: {},
+                InterleaveWays: {},
+                Flags: {}",
+            (*entry).Header.Type,
+            (*entry).Header.Length,
+            (*entry).DeviceHandle,
+            (*entry).DeviceHandle & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 4) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 8) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 12) & ACPI_NFIT_DEVICE_HANDLE_4BIT_MASK,
+            ((*entry).DeviceHandle >> 16) & ACPI_NFIT_DEVICE_HANDLE_12BIT_MASK,
+            (*entry).PhysicalId,
+            (*entry).RegionId,
+            (*entry).RangeIndex,
+            (*entry).RegionIndex,
+            (*entry).RegionSize,
+            (*entry).RegionOffset,
+            (*entry).Address,
+            (*entry).InterleaveIndex,
+            (*entry).InterleaveWays,
+            (*entry).Flags,
+        );
+    }
+}
+
+// NFIT subtable type = 0x2
+fn parse_nfit_interleave_structure(entry: *const ACPI_NFIT_INTERLEAVE) {
+    unsafe {
+        assert_eq!((*entry).Header.Type, 2);
+        debug!(
```

Comment:
Can we just implement debug on type to put actual print logic there?

Reply:
Should be done in the acpica-sys repo?